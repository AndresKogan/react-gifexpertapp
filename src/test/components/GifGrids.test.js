import React from 'react'
import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from '../../Hooks/useFetchGifs';
jest.mock('../../Hooks/useFetchGifs')
describe('Pruebas en <GifGrid/>', () => {

    const category = "one punch";


    test('debe hacer match con el snapshot', () => {

        useFetchGifs.mockReturnValue({

            data:[],
            loading:true

        })


    const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();

    })

    test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {
       
        const gifs = [{
            id:'ABC',
            url: 'https://localhost/cualquiera/cosa.jpg',
            title: 'comunismo'
        }]

        useFetchGifs.mockReturnValue({

            data:gifs,
            loading:false

        })
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })




})