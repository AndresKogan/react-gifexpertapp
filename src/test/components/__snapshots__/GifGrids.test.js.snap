// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__fadeIn"
  >
    one punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="comunismo"
      url="https://localhost/cualquiera/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid/> debe hacer match con el snapshot 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__fadeIn"
  >
    one punch
  </h3>
  <p
    className="card animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
